lnrpc/routerrpc/router_backend.go

```diff
@@ -620,6 +620,7 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 		payIntent.RouteHints = append(
 			payIntent.RouteHints, payReq.RouteHints...,
 		)
+		payIntent.InvoiceFeatures = payReq.Features
```

Comment:
Isn't there an intermediate step before this PR where we still only offer mpp through sendtoroute, but where queryroutes is updated so that it decides whether to mpp or not (probably via an rpc field)? 

Somewhat related: What is the state of payment tracking when the payment is fired through one (or multiple in the case of manual mpp) sendtoroute calls?

Reply:
> Isn't there an intermediate step before this PR where we still only offer mpp through sendtoroute, but where queryroutes is updated so that it decides whether to mpp or not (probably via an rpc field)?

for the moment i'm treating the sendpayment and sendtoroute paths as orthogonal, as the purpose of the PR is have the negotiation of mpp fields be automatic so regular payments can take advantage of the security benefits of mpp. i see pr more as step one towards deprecating legacy payments altogether

i can open another pr that adds this parameter to query routes as well

Reply:
I thought that maybe the queryroutes-only scope would be a subset of this pr. Slowing building up from the totally manual amp case.

>the payment tracking should be complete for single shards in #3499

I realize I asked the wrong question. The tracking is there, but it is not possible to get the raw failure message with `TrackPayment`. It is only possible to obtain that if you keep the sync call to `SendToRoute` open. But if we start exposing the routes and raw failure codes on `TrackPayment`, that is sorted.